### ISSUE TITLE
Pandoc timeout frequently on startup

### ISSUE THREAD

**Issue by foxB612** [wereturtle/ghostwriter#724]
**Platform (please complete the following information):**
 - OS and version [e.g. Ubuntu 18.04, Windows 10, MacOS Sierra]: Arch Linux
 - If Linux, Desktop Environment [e.g., Gnome, Plasma, Cinnamon, etc.]: Plasma

**ghostwriter installation (please complete the following information):**
 - ghostwriter version [see Help->About in menu bar]: 2.1.1
 - If Linux, installation type [e.g., Flatpak, PPA, private build]: pacman community/ghostwriter

**Describe the bug**
I've installed `pandoc` and chosen it for Markdown Flavor in preview options in ghostwriter, however when starting ghostwriter I often get this error message in console: `"pandoc" process timed out and cannot be used.`, and when this error occurs the `pandoc` options will disappear and it'll fall back to `cmark-gfm`. After restarting ghostwriter the `pandoc` may come back hopefully, but next time the error is still likely to happen.

**To Reproduce**
Launch ghostwriter and check the preview options.

**Expected behavior**
`pandoc` should always be available.

**Comment by foxB612** [comment]
Some updates on this issue:

After I switch to [pandoc-bin](https://aur.archlinux.org/packages/pandoc-bin), the error doesn't show up anymore. Therefore I did a simple comparison between pandoc and pandoc-bin: for the **first run** of `pandoc --version` since booting, the result of `time` is:

```
// pandoc
real    0m0.732s
user    0m0.175s
sys     0m0.153s

// pandoc-bin
real    0m0.035s
user    0m0.003s
sys     0m0.008s
```

AFAIK the difference between the two is that pandoc-bin saves the heavy dependency by packaging pandoc into executable binary. I guess it's normal rather than a bug of pandoc?

A dirty fix might be extending the waiting time [here](https://github.com/wereturtle/ghostwriter/blob/c673e6054911d48a014e1c6b7a0516a02f5129c6/src/exporterfactory.cpp#L296-L303) to, like, 1 second or so? But it's not clear to me why timeout is important at starting time: can't we just check the existence of the command (for example `command -v` on linux)? It seems that the issue only exists for the first run of pandoc since booting.

**Comment by wereturtle** [comment]
In the new 2.1.2 version, I have increased the wait time for checking if Pandoc can run.  Please let me know if it is working now or if more tweaking is required.

Thanks!

**Comment by foxB612** [comment]
It seems that the problem still exists.

I see that in the latest code the waiting time of `waitForStarted` is increased. But since in my case the error is `process timed out and cannot be used`, I think the waiting time of `waitForFinished()` should also be increased.

**Comment by git-user314** [comment]
> It seems that the problem still exists.
> 
> I see that in the latest code the waiting time of `waitForStarted` is increased. But since in my case the error is `process timed out and cannot be used`, I think the waiting time of `waitForFinished()` should also be increased.

I agree, that is what I did on version 2.1.1 when I first started to look into this code, as I had the issue with pandoc not being found. On my Windows 10 PC with 2.1.2, pandoc does not show up in the drop-down list.

Here is a link to the commit I did for my quick custom build of 2.1.1: <https://github.com/git-user314/ghostwriter/commit/849759c21b2c6f963a6d240240aabcb1b0199cbf>

I increased the time for `waitForFinished()` from `500` to `1500`, I didn't need to change the time for `waitForStarted()` at all from the original `500` to fix the issue on my PC.
